modules/core/src/main/java/org/apache/ignite/internal/managers/encryption/GridEncryptionManager.java

```diff
@@ -762,21 +1206,95 @@ private void cancelFutures(String msg) {
      *
      * @return {@code true} if local node is coordinator.
      */
-    private boolean notCoordinator() {
+    private boolean isCoordinator() {
         DiscoverySpi spi = ctx.discovery().getInjectedDiscoverySpi();
 
         if (spi instanceof TcpDiscoverySpi)
-            return !((TcpDiscoverySpi)spi).isLocalNodeCoordinator();
+            return ((TcpDiscoverySpi)spi).isLocalNodeCoordinator();
         else {
-            ClusterNode crd = null;
+            ClusterNode crd = U.oldest(ctx.discovery().aliveServerNodes(), null);
 
-            for (ClusterNode node : ctx.discovery().aliveServerNodes()) {
-                if (crd == null || crd.order() > node.order())
-                    crd = node;
-            }
+            return crd != null && F.eq(ctx.localNodeId(), crd.id());
+        }
+    }
+
+    /**
+     * Pending master key.
+     */
+    private static class PendingMasterKey implements Serializable {
```

Comment:
Seems, we should rename this to the `MasterKeyChangeRequest`?

Reply:
Done.